File: src/main/java/com/github/ianiiaannn/operationrelife/handlers/RegistryHandler.java
package com.github.ianiiaannn.operationrelife.handlers;

import com.github.ianiiaannn.operationrelife.blocks.Blocks;
import com.github.ianiiaannn.operationrelife.items.Items;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

public class RegistryHandler {
  private static final IEventBus BUS = FMLJavaModLoadingContext.get().getModEventBus();
  public static void registerAll() {
    Blocks.getRegister().register(RegistryHandler.BUS);
    Items.getRegister().register(RegistryHandler.BUS);
    // Entities.getRegister().register(RegistryHandler.BUS);
  }
}

File: src/main/kotlin/dev/iancmd/operationrelife/handler/RegistryHandler.kt
package dev.iancmd.operationrelife.handler

import dev.iancmd.operationrelife.core.OperationRelife
import dev.iancmd.operationrelife.init.ModBlocks
import dev.iancmd.operationrelife.init.ModDimensions
import dev.iancmd.operationrelife.init.ModFluids
import dev.iancmd.operationrelife.init.ModItems
import net.minecraftforge.fml.common.Mod.EventBusSubscriber
import thedarkcolour.kotlinforforge.forge.MOD_CONTEXT

@EventBusSubscriber(modid = OperationRelife.Companion.MODID)
object RegistryHandler {

  private val BUS = MOD_CONTEXT.getKEventBus()
  fun registerAll() {
    ModBlocks.getRegister().register(BUS)
    ModBlocks.getOverrideRegister().register(BUS)
    ModItems.getRegister().register(BUS)
    ModFluids.getFluidRegister().register(BUS)
    ModFluids.getFluidTypeRegister().register(BUS)
    ModDimensions.dimensionRegistry()
  }
}
